pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -651,6 +651,12 @@ func parseAppInstanceConfig(getconfigCtx *getconfigContext,
 		appinstancePrevConfigHash, configHash, Apps)
 	appinstancePrevConfigHash = configHash
 
+	devUUIDStr, err := os.Hostname()
```

Comment:
Instead of relying on the Hostname being the uuid, why not use the EdgeNodeInfo which the other clustering code uses for this device uuid check?

Reply:
EdgeNodeInfo is parsed and published after parseAppInstanceConfig() so EdgeNodeInfo is not available yet. 

But may I can just do what parseEdgeNodeInfo() is doing to get DeviceID

deviceID, _ := uuid.FromString(config.GetId().Uuid)

Reply:
Makes sense.